apps/apps.humanatlas.io/src/app/pages/us2/services/tissue-origin.service.ts

```diff
@@ -0,0 +1,188 @@
+import { HttpClient } from '@angular/common/http';
+import { inject, Injectable, InjectionToken } from '@angular/core';
+import { firstValueFrom, from, Observable, of, switchMap } from 'rxjs';
+
+/** Sample CSV file URL */
+export const SAMPLE_FILE_URL = 'assets/sample-data/tissue-origin-sample.csv';
+
+/** Injection token for tissue origin predictions API endpoint  */
+export const TISSUE_ORIGIN_ENDPOINT = new InjectionToken<string>('Tissue Origin Endpoint', {
+  providedIn: 'root',
+  factory: () => 'https://apps.humanatlas.io/api/hra-pop',
+});
+
+/**
+ * Supported Tools Data
+ */
+export interface SupportedTools {
+  /** ID with URL */
+  id: string;
+  /** Label */
+  label: string;
+}
+
+/**
+ * User Selection Data
+ */
+export interface UserSelection {
+  /** Organ */
+  selectedOrganIri?: string;
+  /** Tool */
+  selectedToolIri?: string;
+}
+
+/**
+ * Sources Data
+ */
+export interface Sources {
```

Comment:
I might have you improve the typings in hra-api instead of having you redeclare the interfaces here. I will discuss with Bruce and get back to you about what to do.